### PR TITLE
Do some cleanup in the test files and the Lean backend

### PR DESCRIPTION
**Pull request by sonmarcho**

### CODE REVIEW

**Comment by Nadrieril** [comment]
I merged `main` so you'd get the CI changes.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,8 +18,11 @@ permissions:
...
   check_if_skip_duplicate_job:
-    runs-on: [self-hosted, linux, nix]
+    # runs-on: [self-hosted, linux, nix]
+    runs-on: ubuntu-latest
     steps:
+      - uses: DeterminateSystems/nix-installer-action@main
+      - uses: DeterminateSystems/magic-nix-cache-action@main
       - id: skip_check
         uses: fkirc/skip-duplicate-actions@v5
         with:
```
Comment (Nadrieril) [inline .github/workflows/ci.yml:28]: What happened here? Somehow this also took my CI commit x) I think github merge got confused with the merges

Comment (sonmarcho) [reply .github/workflows/ci.yml:28]: This is extremely strange. I never merge other branches than master in my local branch.

Comment (Nadrieril) [reply .github/workflows/ci.yml:28]: I've read before that git merge doesn't deal super well with reverts

Comment (Nadrieril) [reply .github/workflows/ci.yml:28]: Ah no, reverting *merges* is tricky, but reverting individual commit should work :thinking:

Comment (Nadrieril) [reply .github/workflows/ci.yml:28]: Oh, I'm the one who reverted a merge x) All is explained.

**Pull request merged**